src/main/java/ca/corefacility/bioinformatics/irida/ria/web/ajax/projects/ProjectUserGroupsAjaxController.java

```diff
@@ -86,21 +88,40 @@ public ResponseEntity<String> addUserGroupToProject(@RequestParam Long projectId
 	/**
 	 * Update the project role of a user group on the current project
 	 *
+	 * @param projectId   Identifier for a project
+	 * @param id          Identifier for an user group
+	 * @param projectRole Project role to update the user group to
+	 * @param locale      Current users locale
+	 * @return message to user about the result of the update
+	 */
+	@RequestMapping(value = "/role", method = RequestMethod.PUT)
+	public ResponseEntity<String> updateUserGroupRoleOnProject(@RequestParam Long projectId, @RequestParam Long id,
+			String projectRole, Locale locale) {
+		try {
+			return ResponseEntity.ok(service.updateUserGroupRoleOnProject(projectId, id, projectRole, locale));
+		} catch (UIProjectWithoutOwnerException e) {
+			return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+		} catch (UIConstraintViolationException ex) {
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(ex.getErrorMessage());
+		}
+	}
+
+	/**
+	 * Update the project metadata role of a user group on the current project
+	 *
 	 * @param projectId    Identifier for a project
 	 * @param id           Identifier for an user group
-	 * @param role         Role to update the user group to
 	 * @param metadataRole metadata role to update for the user group
 	 * @param locale       Current users locale
 	 * @return message to user about the result of the update
 	 */
-	@RequestMapping(value = "/role", method = RequestMethod.PUT)
-	public ResponseEntity<String> updateUserGroupRoleOnProject(@RequestParam Long projectId, @RequestParam Long id,
-			@RequestParam String role, @RequestParam String metadataRole, Locale locale) {
+	@RequestMapping(value = "/metadata-role", method = RequestMethod.PUT)
+	public ResponseEntity<String> updateUserGroupMetadataRoleOnProject(@RequestParam Long projectId,
+			@RequestParam Long id, String metadataRole, Locale locale) {
 		try {
-			return ResponseEntity.ok(service.updateUserGroupRoleOnProject(projectId, id, role, metadataRole, locale));
-		} catch (ProjectWithoutOwnerException e) {
-			return ResponseEntity.status(HttpStatus.BAD_REQUEST)
-					.body(e.getMessage());
+			return ResponseEntity.ok(service.updateUserGroupMetadataRoleOnProject(projectId, id, metadataRole, locale));
+		} catch (Exception e) {
```

Comment:
We should try and catch the specific exception here, so that we don't unknowingly return a exception message that we don't want to show to the user.

Reply:
which would be `UIProjectWithoutOwnerException` and `UIConstraintViolationException`.

Reply:
This one should not catch a `UIProjectWithoutOwnerException` as changing the metadata role won't change the project role